### PR TITLE
Use json in output

### PR DESCRIPTION
**Pull request by alcortesm**
Merge https://github.com/bblfsh/bash-driver/pull/6 first as the relevant part of this PR is only the commit https://github.com/bblfsh/bash-driver/commit/112ef663c75b8da875a190fa1f2e90b20e1bb6bc.

Solves the second item in https://github.com/src-d/backlog/issues/610 with a very simple AST JSON representation.

Adds a test to check the generated JSON.

### CODE REVIEW

File: native/src/main/java/bblfsh/bash/Main.java [hunk 1]
```diff
@@ -0,0 +1,18 @@
+package bblfsh.bash;
+
+public class Main {
+
+    public static void main(String args[]) {
+        final RequestReader reader = new RequestReader(System.in);
```
Comment (ajnavarro) [inline native/src/main/java/bblfsh/bash/Main.java]: Maybe is better to instance a unique ObjectMapper object and send it to the RequestReader and ReponseWriter over the constructor.

Comment (alcortesm) [inline native/src/main/java/bblfsh/bash/Main.java]: You are right!

Please check out if I have used proper style when injecting the ObjectMapper.

Comment (alcortesm) [inline native/src/main/java/bblfsh/bash/Main.java]: As the ctor of `RequestReader` and `ResponseWriter` is public, maybe I should made `Main.defaultObjectMapper` public too, what do you think?

Comment (ajnavarro) [inline native/src/main/java/bblfsh/bash/Main.java]: I think it is ok like it is, but maybe, to do not have the Main class in tests, you could create a Singleton to provide the mapper.

File: native/src/main/java/bblfsh/bash/ASTNodeSerializer.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+    public ASTNodeSerializer(Class<ASTNode> t) {
+        super(t);
+    }
+
+    @Override
+    public void serialize(ASTNode root, JsonGenerator jG, SerializerProvider provider) throws IOException {
+        jG.writeStartObject();
+
+        final String type = root.getElementType().toString();
```
Comment (smola) [inline native/src/main/java/bblfsh/bash/ASTNodeSerializer.java]: Let's keep names as close as possible to the original (e.g. `elementType` here)

Comment (alcortesm) [inline native/src/main/java/bblfsh/bash/ASTNodeSerializer.java]: Done.

**Comment by smola** [comment]
@alcortesm Please, rebase and clean history.

**Pull request merged**